testdata/src/go.uber.org/errorreturn/inference/errorreturn-with-inference.go

```diff
@@ -176,3 +176,37 @@ func checkAndDeref() {
 		_ = *x //want "error return in position 1 is not guaranteed to be non-nil through all paths"
 	}
 }
+
+// ***** below tests error handling logic for mixed nilable (e.g., pointer) and non-nilable (e.g., string) non-error returns *****
```

Comment:
Nice! Could we add a case for aliased ptr type?

Reply:
Good idea. Added the test case  :)